src/Utils/HashMap.flix

```diff
@@ -0,0 +1,111 @@
+/// Keeps the log computational complexity of Set while removing
+/// the `Order` constraint.
+namespace HashMap {
+    use Hash.hash
+    use HashSet.HashSet
+
+    /// invariants:
+    /// <<A>> There are no mappings with empty lists
```

Comment:
Can we express these as laws?

Reply:
Or maybe laws are public and we need private ones?

Reply:
I don't see any law usage in ex. List. How would you use it here (like syntax and everything, I've never used it :P).

Also, isn't laws on Magnus' chopping block?

Reply:
Actually the first one you can take care of by using a Nel instead of a list.

For `B` you can do something like this I think:

```scala
law noDuplicates: forall(h: HashMap[k, v]) . // some boolean expression that says that all the lists have no duplicates
```

Alternatively you could introduce a `ListSet` type which itself would have that law.